txbuilder/outputs.go

```diff
@@ -25,6 +25,29 @@ type OutputSupplement struct {
 	KeyID string `json:"key_id,omitempty"`
 }
 
+// DustLimit calculates the dust limit
+func DustLimit(outputSize int, feeRate float32) uint64 {
+	dust := float32((outputSize+DustInputSize)*3) * feeRate
+	return uint64(dust)
+}
+
+// DustLimitForOutput calculates the dust limit
+func DustLimitForOutput(output *wire.TxOut, feeRate float32) uint64 {
+	return DustLimit(output.SerializeSize(), feeRate)
+}
+
+// DustLimitForAddress calculates the dust limit
+func DustLimitForAddress(ra bitcoin.RawAddress, feeRate float32) uint64 {
+	lockingScript, err := ra.LockingScript()
+	if err != nil {
+		return 0
```

Comment:
Is the silent error ok?

Reply:
It will only fail when the address is not spendable or is not a known type, but it is probably better to throw and error. I will update it.